docs/source/release_roadmap.org

```diff
@@ -0,0 +1,363 @@
+#+TITLE: Recommended Road
+#+AUTHOR: Aleksandr Petrosyan
+#+DATE: 2023-08-13
+
+* State of the project
+
+Iroha v2 is a capable ledger with notable implemented subroutines that
+allow it to be deployed successfully as a private blockchain ledger:
+1. It has relatively good performance characteristics regarding
+   throughput (post RC9)
+2. It has good performance characteristics regarding transaction
+   finality and latency (post RC19 also query latency).
+3. It has support for Turing complete permission validation, able to
+   enforce arbitrarily complex logic on-chain
+4. It has support for a small interpreted VM called the Iroha Special
+   Instructions which allow small-footprint transactions
+5. It has support for Turing complete smartcontracts living in a
+   sandbox, with a full array of events which can trigger these
+   behaviours.
+6. The ledger has three SDKs within the JVM, JavaScript and Python
+   ecosystems, allowing the users to easily adapt the Iroha node to
+   their requirements. 
+
+
+The ledger has been demonstrated to work during the ADB CSD/RTGS
+linkages project, and is slated to be used in two projects internally
+by Soramitsu which only utilise the private blockchain ledger
+functionality.
+
+The ledger can support un-sharded monolithic blockchains, with limited
+export capability (unlike Iroha v1, v2 uses an internal representation
+and doesn't use a database). Its consensus was designed with Byzantine
+fault tolerance in mind, however, the algorithm -- =sumeragi= has some
+intrinsic flaws outlined in [[https://arxiv.org/pdf/1707.01873.pdf][an IBM Research article]], which is notable
+because IBM research originally designed the B-Chain algorithm which
+serves as a prototype for =sumeragi=.
+
+The blockchain ledger is designed to support a plug-able (and
+upgrade-able) permission validator modules, allowing the blockchain
+invariants to evolve and change with the operation of the blockchain.
+It is capable of resolving simple soft-forks, and has tooling to
+quickly restart a blockchain by squashing all of the information and
+history into a single genesis block. However, it does not retain
+enough flexibility to be used as a single ledger for both private and
+public blockchains. And the following issues have been identified.
+
+1. The more efficient ISI-based language is not usable. The
+   expressions are not sufficiently expressive.
+
+   While this can be somewhat mitigated by the introduction of the
+   DSL, the main expressiveness-boosting facilities of ISI are not
+   usable, and thus only add overhead. A thorough exploration of a
+   restructuring of the Iroha Instruction Set Architecture is needed
+   and shall be published as an RFC in August 2023.
+
+2. WASM is a sub-optimal medium for storage of executable logic. It
+   alone is responsible for 80% of the complexity present in the code
+   base and around 95% of technical debt is related to some mitigation
```

Comment:
Can you please elaborate on this numbers?

Reply:
`no_std`. Much of the dancing around in the `ffi` library has more to do with what is possible on WASM. The fact that the builds are not reproducible can technically be lumped in here as well, given that things like [reproducible central](https://reproducible-builds.org/docs/jvm/) exist for JVM languages.